### DISCUSSION TITLE
Can I login via http and then subscribe to a topic in websockets?

### DISCUSSION THREAD

**Discussion by runandrew** [artilleryio/artillery#1232] [Artillery]
I would like to login a user using an http `POST` request, and then connected to my mqtt broker via websockets using the credentials from the login. Is there a way to do this in one flow?

**Comment by hassy** [comment]
No, not at the moment @runandrew

**Comment by runandrew** [comment]
Thanks.

**Comment by dominik1001** [comment]
I assume this is still not possible, right?

**Comment by hassy** [comment]
@dominik1001 it may be possible to do it with a bit of custom JS via a hook, e.g. [`function`](https://www.artillery.io/docs/guides/guides/ws-reference#running-custom-code-with-function) hook in the WebSocket engine. Is your use case the same as the OP - HTTP login and reuse in WS?

**Reply by madhugod** [reply]
Hello, I would also be able to do HTTP login, then use the token from response in WS

I'm not sure to understand what I should do in the function hook?
Should I just to do the HTTP query myself in the function hook?

**Comment by hassy** [comment]
Converted to a discussion - anyone coming across this via search - the original question may have been asked before Artillery/WS engine supported hooks. :)
